modules_core/com.etendorx.auth/src/main/java/com/etendorx/auth/auth/AuthService.java

```diff
@@ -100,4 +131,21 @@ public static String getOrDefaultValue(String value, String defaultValue) {
     return defaultValue;
   }
 
+  public void validateService(String service, String secret) throws JsonProcessingException {
+    HttpHeaders headers = new HttpHeaders();
+    headers.add("X-TOKEN", token);
+    ResponseEntity<String> rxService = serviceClient.searchServiceByServiceId(service, true, headers);
+    ObjectMapper mapper = new ObjectMapper();
+    ServicesAccessModel servicesAccessModel = mapper.readValue(rxService.getBody(), ServicesAccessModel.class);
+    if (rxService.getStatusCode() != HttpStatus.OK) {
+      throw new ResponseStatusException(rxService.getStatusCode(), "Unsuccessful DAS connection.");
+    }
+    if (servicesAccessModel == null) {
+      throw new ResponseStatusException(HttpStatus.NO_CONTENT, "DAS server respond with not content.");
+    }
+
+    if (servicesAccessModel.get_embedded().getRxServiceses().get(0).getSecret().compareTo(secret) != 0) {
```

Comment:
always exists a Rx servicess ??

Reply:
You must be used StringUtils to compare two strings. Also you can stored in a variable the follow object servicesAccessModel.get_embedded().getRxServiceses().get(0) then you can use it as: 

```
var service = servicesAccessModel.get_embedded().getRxServiceses().get(0);

if (!StringUtils.equals(service.getSearchkey(), reqSearchKey) ) {

if (!StringUtils.equals(service.getSecret(), secret) ) {
```